extensions/iceberg/src/test/java/io/deephaven/iceberg/junit5/SqliteCatalogBase.java

```diff
@@ -283,10 +283,82 @@ void appendMultipleTablesWithDifferentDefinitionTest() {
             tableWriter.append(IcebergWriteInstructions.builder()
                     .addTables(appendTable)
                     .build());
-            failBecauseExceptionWasNotThrown(UncheckedDeephavenException.class);
+            failBecauseExceptionWasNotThrown(TableDefinition.IncompatibleTableDefinitionException.class);
         } catch (TableDefinition.IncompatibleTableDefinitionException e) {
             // Table definition mismatch between table writer and append table
-            assertThat(e).hasMessageContaining("Table definition");
+            assertThat(e).hasMessageContaining("Actual table definition is not compatible with the " +
+                    "expected definition");
+        }
+    }
+
+    @Test
+    void appendWithWrongDefinition() {
+        final Table source = TableTools.emptyTable(10)
+                .update("dateCol = java.time.LocalDate.now()",
+                        "doubleCol = (double) 2.5 * i + 10");
```

Comment:
I'm going to start being a bit stricter with reviews; I'd much prefer if we constructed tables directly instead of using formulas.

Please use `TableTools#newTable(TableDefinition, ColumnHolder<?>...)`; no need for a 10 row table, a 1 row table will suffice. If we find the ergonomics of newTable, xCol lacking, we should want to improve them.